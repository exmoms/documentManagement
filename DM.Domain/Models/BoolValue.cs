namespace DM.Domain.Models
{
    public class BoolValue
    {
        // Forgien-key to metaDataAttribute
        public int MetaDataAttributeId { get; set; }
        public MetaDataAttribute MetaDataAttribute { get; set; }
        // Forgien-key to DocumentVersion
        public int MinDocumentVersionId { get; set; }
        public DocumentVersion DocumentVersion { get; set; }
        public int? MaxDocumentVersionId { get; set; }
        public bool? Value { get; set; }
    }
}
