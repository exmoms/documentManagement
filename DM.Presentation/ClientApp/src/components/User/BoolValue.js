import React, { Component } from "react";
import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel";

class BoolValue extends Component {
  constructor(props) {
    super(props);
    this.state = { value: false };
    this.handleValueChange = this.props.handleValueChange;
  }

  handleChange = (event, checked) => {
    this.setState(
      { value: checked },
      this.handleValueChange(checked, this.props.label)
    );
  };

  render() {
    console.log("check", this.state.value);
    return (
      <div>
        <FormControlLabel
          control={
            <Checkbox
              checked={this.state.value}
              onChange={this.handleChange}
              value="primary"
              inputProps={{ "aria-label": "primary checkbox" }}
            />
          }
          label={this.props.attributeName}
        />
      </div>
    );
  }
}

export default BoolValue;
