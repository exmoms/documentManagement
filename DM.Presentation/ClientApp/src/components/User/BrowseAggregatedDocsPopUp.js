/**
 * @file Contains BrowseAggregatedDocsPopUp component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import React, { Component } from "react";
import DocumentTable from "./DocumentTable";
import SearchDocuments from "./SearchDocuments";
import CREATE_DOC_STYLES from "./Styles";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContentText from "@material-ui/core/DialogContentText";

import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Paper from "@material-ui/core/Paper";

import { withTranslation } from "react-i18next";

const styles = CREATE_DOC_STYLES;

class BrowseAggregatedDocsPopUp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      obj: {
        id: -1,
        name: "...",
        latest_version: -1,
      },
      data: [],
      columns: [],
    };
    this.updateParentState = this.props.action;
  }
  t = this.props["t"];

  getDocumentList = (list, columns) => {
    this.setState({
      data: list,
      columns: columns,
    });
  };

  toggleShow = (show) => {
    this.updateParentState("show", show);
  };

  handleSelect = (value) => {
    if (value !== null) {
      this.updateParentState("addDoc", {
        modelId: this.props.modelId,
        selectedDoc: value,
      });

      this.toggleShow(false);
    }
  };

  /**
   * @todo Could we reduce rendering this component, once it renders two times
   * each time we update state
   */
  render() {
    const { classes } = this.props;
    return (
      <div>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.toggleShow(false)}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title">
            {this.t("addaggregateddoc")}
          </DialogTitle>
          <DialogContent>
            <DialogContentText>
              {this.t("dialog_selectaggregateddocument")} <strong>Hint:</strong>{" "}
              {this.t("dialog_previewchooseddocument")}
            </DialogContentText>
            <TableContainer component={Paper}>
              <Table className={classes.table} aria-label="simple table">
                <TableBody>
                  <TableRow key="search-form">
                    <TableCell>
                      <SearchDocuments
                        getDocumentList={this.getDocumentList}
                        action="update-document"
                        modelId={this.props.modelId}
                      />
                      <DocumentTable
                        data={this.state.data}
                        columns={this.state.columns}
                        action="update-document"
                        handleSelect={this.handleSelect}
                      />
                    </TableCell>
                  </TableRow>
                </TableBody>
              </Table>
            </TableContainer>
          </DialogContent>
          <DialogActions>
            <Button
              id="CancelButton"
              onClick={() => this.toggleShow(false)}
              color="primary"
            >
              {this.t("button_cancel")}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(BrowseAggregatedDocsPopUp));
