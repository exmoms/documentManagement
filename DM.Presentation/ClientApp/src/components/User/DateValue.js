import React, { Component } from "react";
import "date-fns";
import DateFnsUtils from "@date-io/date-fns";
import {
  KeyboardDatePicker,
  MuiPickersUtilsProvider,
} from "@material-ui/pickers";
import { withTranslation } from "react-i18next";

class DateValue extends Component {
  constructor(props) {
    super(props);
    this.state = { date: null };
    this.handleValueChange = this.props.handleValueChange;
  }

  t = this.props["t"];

  handleChange = (date, value) => {
    this.setState(
      { date: date },
      this.handleValueChange(date, this.props.label)
    );
  };

  render() {
    return (
      <div>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <KeyboardDatePicker
            margin="normal"
            id={this.props.label}
            label={this.t(this.props.label)}
            format="MM/dd/yyyy"
            value={this.state.date}
            onChange={this.handleChange}
            KeyboardButtonProps={{
              "aria-label": "change date",
            }}
          />
        </MuiPickersUtilsProvider>
      </div>
    );
  }
}

export default withTranslation()(DateValue);
