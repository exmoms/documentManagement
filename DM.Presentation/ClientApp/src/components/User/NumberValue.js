import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import { withTranslation } from "react-i18next";

class NumberValue extends Component {
  constructor(props) {
    super(props);
    this.state = { value: 0 };
    this.handleValueChange = this.props.handleValueChange;
  }

  t = this.props["t"];

  handleChange = (event) => {
    let value = parseInt(event.target.value);
    this.setState(
      { value: value },
      this.handleValueChange(value, this.props.label)
    );
  };

  render() {
    return (
      <div>
        <TextField
          id={this.props.label}
          label={this.t(this.props.label)}
          type="number"
          onChange={this.handleChange}
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>
    );
  }
}

export default withTranslation()(NumberValue);
