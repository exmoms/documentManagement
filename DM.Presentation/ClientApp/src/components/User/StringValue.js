import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import { withTranslation } from "react-i18next";

class StringValue extends Component {
  constructor(props) {
    super(props);
    this.state = { text: "" };
    this.handleValueChange = this.props.handleValueChange;
  }

  t = this.props["t"];

  handleChange = (event) => {
    let value = event.target.value;
    this.setState(
      { text: value },
      this.handleValueChange(value, this.props.label)
    );
  };

  render() {
    return (
      <div>
        <TextField
          id={this.props.label}
          label={this.t(this.props.label)}
          variant="outlined"
          onChange={this.handleChange}
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>
    );
  }
}

export default withTranslation()(StringValue);
