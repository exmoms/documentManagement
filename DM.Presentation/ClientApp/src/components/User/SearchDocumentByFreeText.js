import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Autocomplete from "@material-ui/lab/Autocomplete";
import React, { Component } from "react";
import { withTranslation } from "react-i18next";
import { fetchData } from "../../api/FetchData";
import StringValue from "./StringValue";
import styled from "styled-components";

class SearchDocumentByFreeText extends Component {
  constructor() {
    super();
    this.state = {
      model_id: -1,
      model_label: "",
      last_model_id: -1,
      model_list: [],
      class_id: -1,
      class_label: "",
      last_class_id: -1,
      class_list: [],
      text: "",
    };
  }

  componentDidMount() {
    fetchData("/api/MetaDataModel/GetMetaDataModelsIdName")
      .then((value) => {
        const data = value;
        const lista = [];
        data.forEach((item) =>
          lista.push({
            id: item.id,
            title: item.metaDataModelName,
            selected: false,
          })
        );
        this.setState({
          model_list: lista,
        });
      })
      .catch((e) => console.log(e));

    fetchData("/api/DocumentClass/")
      .then((value) => {
        const list = [];
        value.forEach((item) =>
          list.push({
            id: item.id,
            title: item.documentClassName,
            selected: false,
          })
        );
        this.setState({
          class_list: list,
        });
      })
      .catch((e) => console.log(e));
  }

  componentDidUpdate() {
    if (
      typeof this.props.modelId !== "undefined" &&
      this.state.model_id === -1
    ) {
      const model = this.state.model_list.find(
        (item) => this.props.modelId === item.id
      );
      if (typeof model !== "undefined") {
        this.setState({
          model_list: [model],
          model_id: model.id,
          model_label: model.title,
          class_id: -1,
          class_label: "",
        });
      }
    }
  }

  handleModelChange = (value) => {
    if (typeof this.props.modelId === "undefined") {
      if (value !== null) {
        this.setState({
          model_id: value.id,
          model_label: value.title,
          class_id: -1,
          class_label: "",
        });
      } else {
        this.setState({
          class_id: -1,
          class_label: "",
          model_id: -1,
          model_label: "",
        });
      }
    }
  };

  handleClassChange = (value) => {
    if (value !== null) {
      this.setState({
        class_id: value.id,
        class_label: value.title,
        model_id: -1,
        model_label: "",
      });
    } else {
      this.setState({
        class_id: -1,
        class_label: "",
        model_id: -1,
        model_label: "",
      });
    }
  };

  handleChangeInput = (value) => {
    this.setState({ text: value });
  };

  addDefaultModel = (model_id) => {
    let result = { title: "", id: -1 };
    if (typeof model_id !== "undefined") {
      const model = this.state.model_list.find((item) => item.id === model_id);
      if (typeof model !== "undefined") {
        result.title = model.title;
        result.id = model.id;
      } else {
        // error message
        console.log("[ERROR] The Model is not found");
      }
    } else {
      result.title = this.state.model_label;
      result.id = this.state.model_id;
    }
    return result;
  };

  handleSearch = () => {
    let payload = {
      text: null,
      modelId: null,
      classId: null,
      setId: null,
      setIdExclude: null,
    };

    let api_url = "/api/document/SearchByFreeTxt";
    payload.text = this.state.text;

    let { model_id, class_id } = this.state;
    if (model_id !== -1) {
      payload.modelId = model_id;
    }
    if (class_id !== -1) {
      payload.classId = class_id;
    }

    if (this.props.setIdExclude !== undefined) {
      payload.setIdExclude = this.props.setIdExclude;
    }
    this.props.composeSearchRequest(api_url, payload);
  };

  render() {
    const model_id_exist = typeof this.props.modelId !== "undefined";
    const { t } = this.props;
    return (
      <Container>
        <InnerContainer>
          <Autocomplete
            id="combo-box-model-list"
            onChange={(event, value) => this.handleModelChange(value)}
            options={this.state.model_list}
            getOptionLabel={(option) => option.title}
            style={{ width: 300 }}
            value={this.addDefaultModel(this.props.modelId)}
            renderInput={(params) => (
              <TextField
                {...params}
                label={t("createdocument_meatdatmodel")}
                variant="outlined"
              />
            )}
          />
        </InnerContainer>
        <InnerContainer>
          <Autocomplete
            id="combo-box-class-id"
            onChange={(event, value) => this.handleClassChange(value)}
            options={this.state.class_list}
            getOptionLabel={(option) => option.title}
            style={{ width: 300 }}
            value={{ title: this.state.class_label, id: this.state.class_id }}
            disabled={!!model_id_exist}
            renderInput={(params) => (
              <TextField
                {...params}
                label={t("label_documentclass")}
                variant="outlined"
              />
            )}
          />
        </InnerContainer>
        <InnerContainer>
          <StringValue
            label={t("label_textvalue")}
            handleValueChange={this.handleChangeInput}
          />
        </InnerContainer>
        <InnerContainer>
          <Button
            variant="contained"
            color="primary"
            onClick={this.handleSearch}
            id="search-document-button"
          >
            {t("buttton_search")}
          </Button>
        </InnerContainer>
      </Container>
    );
  }
}

export default withTranslation()(SearchDocumentByFreeText);

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const InnerContainer = styled.div`
  marign-top: 20px;
  margin-bottom: 20px;
`;
