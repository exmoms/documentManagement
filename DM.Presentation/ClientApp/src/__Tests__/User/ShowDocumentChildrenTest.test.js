import React from "react";
import ReactDom from "react-dom";
import SearchDocumentByFreeText from "../../components/User/SearchDocumentByFreeText";
import BrowseAggregatedDocsPopUp from "../../components/User/BrowseAggregatedDocsPopUp";
import ShowDocumentChildren from "../../components/User/ShowDocumentChildren";
import DateValue from "../../components/User/DateValue";
import BoolValue from "../../components/User/BoolValue";
import NumberValue from "../../components/User/NumberValue";
import StringValue from "../../components/User/StringValue";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import Button from "@material-ui/core/Button";
afterEach(cleanup);
describe("ShowDocumentChildren", () => { 
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A"}];
    let option=true;
    const div = document.createElement("div");
    ReactDom.render(
      <ShowDocumentChildren
      updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}>
      </ShowDocumentChildren>,
      div
    );
  });
  
  test("ShowDocumentChildren matches snapshot", () => {
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A"}];
    let option=true;
    const component = renderer.create(
        <ShowDocumentChildren
        updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}>
        </ShowDocumentChildren>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });  

  test("intializations",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    expect(wrapper.find(TableRow).length).toBe(2);
    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
});

test("changeChildDocOnClickHandler",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    //calling handleChange
   act(() => {
    wrapper
      .find(TableCell).find(Button)
      .at(0)
      .props()
      .onClick();
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeTruthy();
});

test("updateChildDocOnClickHandler",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    const consoleMock = jest.spyOn(console, "log")
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    //calling handleChange
   act(() => {
    wrapper
      .find(TableCell).find(Button)
      .at(1)
      .props()
      .onClick();
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
    expect(updateDocument).toHaveBeenCalledTimes(1);
    expect(updateDocument.mock.calls[0][0]).toBe(2);

    expect(consoleMock).toHaveBeenCalledTimes(1);
    expect(consoleMock.mock.calls[0][0]).toBe("form ShowDocumentChildern:");
    expect(consoleMock.mock.calls[0][1]).toBe(2);
});

test("updateStateFromChild when (attribName === show) ",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
    let attribName="show";
    let newState=true;
    //calling updateStateFromChild
   act(() => {
    wrapper
      .find(BrowseAggregatedDocsPopUp)
      .at(0)
      .props()
      .action(attribName,newState);
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeTruthy();
});

test("updateStateFromChild when (attribName === addDoc) and (index === -1)  and(process.env.NODE_ENV === development)",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    const consoleMock = jest.spyOn(console, "log")
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    let attribName="addDoc";
    let newState={modelId:2,selectedDoc:{id:1}};
    process.env.NODE_ENV="development";
    //calling updateStateFromChild
   act(() => {
    wrapper
      .find(BrowseAggregatedDocsPopUp)
      .at(0)
      .props()
      .action(attribName,newState);
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
    expect(consoleMock).toHaveBeenCalled();
    //unexpected behaviour according to the compiler
    // expect(consoleMock.mock.calls[0][0]).toBe("[ERROR] Can't find the newState.modelId:");
    // expect(consoleMock.mock.calls[0][1]).toBe(2);
});

test("updateStateFromChild when (attribName === addDoc) and (index === -1)  and(process.env.NODE_ENV !== development)",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    const consoleMock = jest.spyOn(console, "log")
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    let attribName="addDoc";
    let newState={modelId:2,selectedDoc:{id:1}};
    process.env.NODE_ENV="";
    //calling updateStateFromChild
   act(() => {
    wrapper
      .find(BrowseAggregatedDocsPopUp)
      .at(0)
      .props()
      .action(attribName,newState);
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
    expect(consoleMock).toHaveBeenCalledTimes(0);
});

test("updateStateFromChild when (attribName === addDoc) and (index !== -1) and (process.env.NODE_ENV === development)",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    const consoleMock = jest.spyOn(console, "log")
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    let attribName="addDoc";
    let newState={modelId:1,selectedDoc:{id:1}};
    process.env.NODE_ENV="development";
    //calling updateStateFromChild
   act(() => {
    wrapper
      .find(BrowseAggregatedDocsPopUp)
      .at(0)
      .props()
      .action(attribName,newState);
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
    expect(consoleMock).toHaveBeenCalled();
    //unexpected behaviour according to the compiler
    // expect(consoleMock.mock.calls[0][0]).toBe("[SUCCESS] Update with the newState.modelId");
    // expect(consoleMock.mock.calls[0][1]).toBe(1);
    // expect(consoleMock.mock.calls[0][2]).toBe("with selected id:");
    // expect(consoleMock.mock.calls[0][3]).toBe(1);

    expect(updater).toHaveBeenCalledTimes(1);
    expect(updater.mock.calls[0][0]).toStrictEqual([{"aggregateName": "aggregateName A", "childDocumentVersionId": undefined, "childMetadataModelId": 1, "documentName": undefined}]);
});

test("updateStateFromChild when (attribName === addDoc) and (index !== -1) and (process.env.NODE_ENV !== development)",  () =>{
    let updater = jest.fn();
    let updateDocument = jest.fn();
    let documents=[{childMetadataModelId:1,childDocumentVersionId:2,documentName:"documentName A" , aggregateName:"aggregateName A" , documentName:"documentName A"}];
    let option=true;
    const consoleMock = jest.spyOn(console, "log")
    let wrapper = mount(
      <ShowDocumentChildren updater ={updater} updateDocument ={updateDocument} documents ={documents} option ={option}/>
    );
    let attribName="addDoc";
    let newState={modelId:1,selectedDoc:{id:1}};
    process.env.NODE_ENV="";
    //calling updateStateFromChild
   act(() => {
    wrapper
      .find(BrowseAggregatedDocsPopUp)
      .at(0)
      .props()
      .action(attribName,newState);
  });
  wrapper.update();
  wrapper.render();

    expect(wrapper.find(TableBody).find(TableCell).at(0).text()).toBe("aggregateName A");
    expect(wrapper.find(TableBody).find(TableCell).at(1).text()).toBe("documentName A");
    expect(wrapper.find(TableCell).find(Button).at(0).props().disabled).toBeFalsy();
    expect(wrapper.find(TableCell).find(Button).at(1).props().disabled).toBeFalsy();

    expect(wrapper.find(BrowseAggregatedDocsPopUp).length).toBe(1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().modelId).toBe(-1);
    expect(wrapper.find(BrowseAggregatedDocsPopUp).at(0).props().show).toBeFalsy();
    expect(consoleMock).toHaveBeenCalledTimes(0);

    expect(updater).toHaveBeenCalledTimes(1);
    expect(updater.mock.calls[0][0]).toStrictEqual([{"aggregateName": "aggregateName A", "childDocumentVersionId": undefined, "childMetadataModelId": 1, "documentName": undefined}]);
});
}); 