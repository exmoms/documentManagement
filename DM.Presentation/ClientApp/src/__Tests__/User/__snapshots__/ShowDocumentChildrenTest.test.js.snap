// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowDocumentChildren ShowDocumentChildren matches snapshot 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="simple table"
        className="MuiTable-root withI18nextTranslation(ShowDocumentChildren)-table-6"
      >
        <thead
          className="MuiTableHead-root"
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              tablecell_aggregatedname
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              tablecell_documentname
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              tablecell_actions
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
        >
          <tr
            className="MuiTableRow-root"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              documentName A
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined withI18nextTranslation(ShowDocumentChildren)-buttonTable-11 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  button_change
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-outlined withI18nextTranslation(ShowDocumentChildren)-buttonTable-11 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  button_update
                </span>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div />
  </div>
</div>
`;
